modules/rtbhouseBidAdapter.js

```diff
@@ -96,30 +63,257 @@ export const spec = {
     };
   },
   interpretResponse: function (serverResponse, originalRequest) {
-    serverResponse = serverResponse.body;
-    const bids = [];
-
-    if (utils.isArray(serverResponse)) {
-      serverResponse.forEach(serverBid => {
-        if (serverBid.price !== 0) {
-          const bid = {
-            requestId: serverBid.impid,
-            mediaType: BANNER,
-            cpm: serverBid.price,
-            creativeId: serverBid.adid,
-            ad: serverBid.adm,
-            width: serverBid.w,
-            height: serverBid.h,
-            ttl: 55,
-            netRevenue: true,
-            currency: 'USD'
-          };
-          bids.push(bid);
-        }
-      });
+    const responseBody = serverResponse.body;
+    if (!utils.isArray(responseBody)) {
+      return [];
     }
+
+    const bids = [];
+    responseBody.forEach(serverBid => {
+      if (serverBid.price === 0) {
+        return;
+      }
+      if (serverBid.adm.startsWith('{"native')) {
+        bids.push(interpretNativeBid(serverBid));
+      } else {
+        bids.push(interpretBannerBid(serverBid));
+      }
+    });
     return bids;
   }
 };
 
+function buildEndpointUrl(region) {
+  return 'https://' + region + '.' + ENDPOINT_URL;
+}
+
+/**
+ * Produces OpenRTB Imp object from a slot config
+ */
+function mapImpression(slot) {
+  return {
+    id: slot.bidId,
+    banner: mapBanner(slot),
+    native: mapNative(slot),
+    tagid: slot.adUnitCode.toString()
+  };
+}
+
+/**
+ * Produces OpenRTB Banner object
+ */
+function mapBanner(slot) {
+  if (slot.mediaType === 'banner' ||
+    utils.deepAccess(slot, 'mediaTypes.banner') ||
+    (!slot.mediaType && !slot.mediaTypes)) {
+    return {
+      w: slot.sizes[0][0],
+      h: slot.sizes[0][1],
+      format: slot.sizes.map(size => ({
+        w: size[0],
+        h: size[1]
+      }))
+    };
+  }
+}
+
+/**
+ * Produces an OpenRTB Site object
+ */
+function mapSite(validRequest) {
+  const pubId = validRequest && validRequest.length > 0
+    ? validRequest[0].params.publisherId : 'unknown';
+  return {
+    publisher: {
+      id: pubId.toString(),
+    },
+    page: utils.getTopWindowUrl(),
+    name: utils.getOrigin()
+  }
+}
+
+/**
+ * Produces an OpenRTB Native object from a slot config
+ */
+function mapNative(slot) {
+  if (slot.mediaType === 'native' || utils.deepAccess(slot, 'mediaTypes.native')) {
+    return {
+      request: {
+        assets: mapNativeAssets(slot)
+      },
+      ver: '1.1'
+    }
+  }
+}
+
+/**
+ * Produces an OpenRTB Native Ads Assets objects
+ */
+function mapNativeAssets(slot) {
+  const params = slot.nativeParams || utils.deepAccess(slot, 'mediaTypes.native');
+  const assets = [];
+  if (params.title) {
+    assets.push({
+      id: OPENRTB.NATIVE.ASSET_ID.TITLE,
+      required: params.title.required ? 1 : 0,
+      title: {
+        len: params.title.len || 140
```

Comment:
why default to 140?

Reply:
In OpenRTB len param is `required` and in Prebid it's not so it needs default value, 140 because in OpenRTB they recommend title len to be: _"25, 90, or 140"_ so I choose the highest value. Maybe it's not the wisest choice, is it?

Reply:
I'd go with 25 if nothing else is defined. This is standard for Google Native, so probably good to go with this way

Reply:
Fixed.